vcd/resource_vcd_nsxt_alb_edgegateway_service_engine_group.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		return diag.Errorf("error deleting ALB Service Engine Group assignment to Edge Gateway: %s", err)
+	}
+	return nil
+}
+
+func resourceVcdAlbEdgeGatewayServiceEngineGroupImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] NSX-T ALB Service Engine Group assignment import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.vdc-name.nsxt-edge-gw-name.se-group-name")
+	}
+	orgName, vdcName, edgeName, seGroupName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find Org %s: %s", vdcName, err)
+	}
+
+	if vdc.IsNsxv() {
+		return nil, fmt.Errorf("this resource is only supported for NSX-T Edge Gateways")
+	}
+
+	edge, err := vdc.GetNsxtEdgeGatewayByName(edgeName)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve NSX-T Edge Gateway with ID '%s': %s", d.Id(), err)
+	}
+
+	seGroupAssignment, err := vcdClient.GetAlbServiceEngineGroupAssignmentByName(seGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("errorr retrieving Servce Engine Group assignment to Edge Gateway with Name '%s': %s",
+			seGroupName, err)
+	}
+
+	dSet(d, "edge_gateway_id", edge.EdgeGateway.ID)
+	d.SetId(seGroupAssignment.NsxtAlbServiceEngineGroupAssignment.ID)
+	return []*schema.ResourceData{d}, nil
+}
+
+func setAlbServiceEngineGroupAssignmentData(d *schema.ResourceData, t *types.NsxtAlbServiceEngineGroupAssignment) {
+	dSet(d, "edge_gateway_id", t.GatewayRef.ID)
+	dSet(d, "service_engine_group_id", t.ServiceEngineGroupRef.ID)
+	dSet(d, "max_virtual_services", t.MaxVirtualServices)
+	dSet(d, "reserved_virtual_services", t.MinVirtualServices)
+	dSet(d, "deployed_virtual_services", t.NumDeployedVirtualServices)
+}
+
+func getAlbServiceEngineGroupAssignmentType(d *schema.ResourceData) *types.NsxtAlbServiceEngineGroupAssignment {
+	edgeAlbServiceEngineAssigmentConfig := &types.NsxtAlbServiceEngineGroupAssignment{
+		GatewayRef:            &types.OpenApiReference{ID: d.Get("edge_gateway_id").(string)},
+		ServiceEngineGroupRef: &types.OpenApiReference{ID: d.Get("service_engine_group_id").(string)},
+	}
+
+	// Max Virtual Services and Reserved Virtual Services only work with SHARED Service Engine Group, but validation
+	// enforcement is left for VCD API.
+	if maxServicesInterface, isSet := d.GetOk("max_virtual_services"); isSet {
+		edgeAlbServiceEngineAssigmentConfig.MaxVirtualServices = takeIntPointer(maxServicesInterface.(int))
+	}
+
+	if reservedServicesInterface, isSet := d.GetOk("reserved_virtual_services"); isSet {
+		edgeAlbServiceEngineAssigmentConfig.MinVirtualServices = takeIntPointer(reservedServicesInterface.(int))
+	}
+
+	return edgeAlbServiceEngineAssigmentConfig
+}
```

Comment:
It's just me a little lost. Is AlbServiceEngineGroup or AlbServiceEngineGroupAssigment. In govcd I see AlbServiceEngineGroupAssigment in terraform I see both things. Are they different or same thing with two names?

Reply:
The Assignment word in govcd comes from API  (loadBalancer/serviceEngineGroups/assignments/) however VCD UI does not use the term Assignment. The idea is that govcd follows API, while in Terraform we follow UI naming. In this case this was a leftover variable name from initial coding. I have removed the 'assignment' word.